apis/core.oam.dev/common/types.go

```diff
@@ -198,7 +208,7 @@ type AppStatus struct {
 	Phase ApplicationPhase `json:"status,omitempty"`
 
 	// Components record the related Components created by Application Controller
-	Components []v1alpha12.TypedReference `json:"components,omitempty"`
+	Components []ComponentRef `json:"components,omitempty"`
 
 	// Services record the status of the application services
 	Services []ApplicationComponentStatus `json:"services,omitempty"`
```

Comment:
does the information already contained in `Services`?  Is it the name field in it?

https://github.com/oam-dev/kubevela/issues/1351

Reply:
Let me check.

Reply:
I think `Services` just record the name of workload such as `webservice`, what we want to add here is the type of `workloadDefinition` such as `deployments.app`.

Ref:
https://github.com/oam-dev/kubevela/blob/63b348f4d887bf522d476f0614de112293e7ead3/pkg/controller/core.oam.dev/v1alpha2/application/apply.go#L197-L204

Reply:
In the issue #1351  we just want to know the instance name, I think the name in service is enough

Reply:
I don't think so, according to the isse #1351 , `WORKLOAD-KIND` is actually wanted in my opinion.
![image](https://user-images.githubusercontent.com/6824860/114504969-3da6c600-9c62-11eb-8fde-61c49d1030ee.png)

The name is already exist, if you think `WORKLOAD-KIND` is not necessary, I will close this PR and I think the issue #1351 could be closed.

Reply:
If so, I think we could add the workloadDef info into `ApplicationComponentStatus` struct

Reply:
ok, I will try.